src/HTTPCommands.cc

```diff
@@ -380,6 +427,8 @@ bool HTTPRequest::sendPreparedRequest(const std::string &protocol,
 		}
 		return false;
 	}
+	// rv = curl_easy_setopt(curl.get(), CURLOPT_DEBUGFUNCTION, debug_callback);
```

Comment:
- Pass the logging object to the debug function and use the logging facility instead.
- Hook the request logging into the configuration subsystem (take a look into https://github.com/PelicanPlatform/xrootd-s3-http/blob/main/src/logging.cc; perhaps add a new `dump` or `trace` level?).

Reply:
I understand the second bullet and can make that happen. Also, yes, it almost has to be a new level because this will result in crazy large logfiles and probably a bit of latency. 

I don't think I get the first bullet though. We are using a libcurl call here to register the debug callback. How would we pass a logging object into that library and have it be available when the callbacks are made?